### PR TITLE
Add QuestDB Output Component

### PR DESCRIPTION
**Pull request by sklarsa**
Adds an output component for writing to [QuestDB](https://questdb.io/) tables using ILP (Influx Line Protocol) as implemented by the official [QuestDB Go Client](https://github.com/questdb/go-questdb-client).

### CODE REVIEW

**Comment by emaxerrno** [comment]
This is cool

File: internal/impl/questdb/output.go [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+			service.NewBoolField("tlsEnabled").
+				Description("Use TLS to secure the connection to the server").
+				Optional(),
+			service.NewStringField("tlsVerify").
+				Description("Whether to verify the server's certificate. This should only be used for testing as a "+
+					"last resort and never used in production as it makes the connection vulnerable to "+
+					"man-in-the-middle attacks. Options are 'on' or 'unsafe_off'.").
+				Optional().
+				Default("on").
+				LintRule(`root = if ["on","unsafe_off"].contains(this != true { ["valid options are \"on\" or \"unsafe_off\"" ] }`),
```
Comment (rockwotj) [inline internal/impl/questdb/output.go:62]: We have a service.NewTLSToggledField("tls") that has default fields supported everywhere - can we please us that?

https://github.com/redpanda-data/connect/blob/1c664da8ab51b796bb5e11189fdd41f2b41d9125/internal/impl/kafka/input_kafka_franz.go#L123

File: internal/impl/questdb/output.go [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+			service.NewDurationField("retryTimeout").
```
Comment (rockwotj) [inline internal/impl/questdb/output.go:63]: We use snake case naming: `retry_timeout`. This also applies to other fields.

File: internal/impl/questdb/output.go (hunk 1 above)
Comment (sklarsa) [reply internal/impl/questdb/output.go]: Yep, how's this? Our sender uses the tlsConfig contained in the unmarshalled conf.

**Review by rockwotj — APPROVED** [review]
I'm not too familiar with QuestDB, but the code LGTM. Will let other team members review as well.

**Comment by Jeffail** [comment]
LGTM, thanks @sklarsa!

**Pull request merged**